### PR TITLE
core: frontend: version-chooser: refactor docker pull output

### PR DESCRIPTION
**Pull request by Williangalvani**
This will be used by the Kraken aswell

### CODE REVIEW

File: core/frontend/src/utils/pull_tracker.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    layers: string[] = []
+
+    layer_progress: Dictionary<string> = {}
+
+    overall_status = ''
+
+    left_over_data = ''
+
+    ready_to_show = false
```
[the remark this answers is not in the document]
Comment (Williangalvani) [reply core/frontend/src/utils/pull_tracker.ts:14]: nice. I mark the other ones private as there's an eslint rule for unused private variables

File: core/frontend/src/utils/pull_tracker.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    onready: () => any = () => null
```
[the remark this answers is not in the document]
Comment (Williangalvani) [reply core/frontend/src/utils/pull_tracker.ts]: done

File: core/frontend/src/utils/pull_tracker.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    private layers: string[] = []
+
+    private layer_progress: Dictionary<string> = {}
+
+    private left_over_data = ''
+
+    pull_output = ''
+
+    private onready: () => any
```
Comment (patrickelectric) [inline core/frontend/src/utils/pull_tracker.ts:14]: Why the return type is any and not void ?

Comment (Williangalvani) [reply core/frontend/src/utils/pull_tracker.ts]: because I forgot what Ts used for void. oops.

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**